lib/rspec/core/metadata_filter.rb

```diff
@@ -15,22 +15,19 @@ def apply?(predicate, filters, metadata)
         # @private
         def filter_applies?(key, value, metadata)
           silence_metadata_example_group_deprecations do
-            return filter_applies_to_any_value?(key, value, metadata) if Array === metadata[key] && !(Proc === value)
             return location_filter_applies?(value, metadata)          if key == :locations
             return id_filter_applies?(value, metadata)                if key == :ids
             return filters_apply?(key, value, metadata)               if Hash === value
 
             return false unless metadata.key?(key)
+            return true if value.is_a?(TrueClass) && !!metadata[key]
```

Comment:
The bigger concern here IMO is the way `filter_applies_to_any_value?` flips the metadata and filter values when looping back to `filter_applies?`.  This doesn't seem right, but I didn't modify it at all because it seemed risky.  Most cases I could think of though, this doesn't seem to matter, as it should still only match if someone is filtering an array using `before(:foo)` or `before(foo: true)` and there _happens_ to be an array on an example group with the same metadata name and a literal `true` as a member of the array, e.g. `it 'does something', foo: [true] do` but even so, I'm 99.9% certain this would have been previously matched by the string comparison at the bottom of `filter_applies?` so I think this is still not actually a change in behavior.